src/main/java/core/basesyntax/supplier/ShapeSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.shape.Circle;
+import core.basesyntax.shape.IsoscelesTrapezoid;
+import core.basesyntax.shape.Rectangle;
+import core.basesyntax.shape.RightTriangle;
+import core.basesyntax.shape.Shape;
+import core.basesyntax.shape.Square;
+import core.basesyntax.type.Color;
+
+public class ShapeSupplier extends BasicRandomValueSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    @Override
+    public Shape getRandom() {
+        Color color = colorSupplier.getRandom();
+        double randomDouble1 = randomGenerator.nextDouble();
+        double randomDouble2 = randomGenerator.nextDouble();
+        double randomDouble3 = randomGenerator.nextDouble();
+
+        int randomShapePick = randomGenerator.nextInt(Shape.SHAPES_COUNT);
```

Comment:
Replace Shape.SHAPES_COUNT with an integer constant representing the number of shapes.